### PR TITLE
samples: Add sample snippets for IN, NOT_EQUALS, and NOT_IN

### LINKED ISSUE #738: Add documentation snippets for IN/NOT_IN/NOT_EQUAL operators
I work on the Firestore/Datastore documentation, and I'd like to add some example snippets for IN/NOT_IN/NOT_EQUAL operators.

File to update:
[samples/snippets/src/test/java/com/google/datastore/snippets/ConceptsTest.java](https://github.com/googleapis/java-datastore/blob/main/samples/snippets/src/test/java/com/google/datastore/snippets/ConceptsTest.java)

These snippets will depend on #688.

### PR DESCRIPTION
**Pull request by jlara310**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #738 ☕️

Depends on #688 
If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

**Comment by jlara310** [comment]
Now that #688 is merged, this is also ready for review/merge.

**Comment by jlara310** [comment]
The project used for testing requires the index below for one of the snippets:

```
- kind: Task
  properties:
  - name: tag
  - name: tag
 ```
 
 Can someone help me add this index to the testing project?

File: samples/snippets/src/test/java/com/google/datastore/snippets/ConceptsTest.java [hunk 1]
```diff
@@ -1044,4 +1044,81 @@ public void testPropertyFilteringRunQuery() {
...
+  public void testEqQuerySorted() {
+    setUpQueryTests();
+    // [START datastore_eq_query_sorted]
+    Query<Entity> query =
+        Query.newEntityQueryBuilder()
+            .setKind("Task")
+            .setFilter(
+                PropertyFilter.eq("tag", "learn")
+            )
+            .setOrderBy(OrderBy.asc("tag"))
```
Comment (kolea2) [inline samples/snippets/src/test/java/com/google/datastore/snippets/ConceptsTest.java:1103]: to confirm, should we also be adding a not in sorted example?

**Review by kolea2 — APPROVED** [review]
LGTM with one question. Thanks!

File: samples/snippets/src/test/java/com/google/datastore/snippets/ConceptsTest.java (hunk 1 above)
Comment (jlara310) [reply samples/snippets/src/test/java/com/google/datastore/snippets/ConceptsTest.java:1103]: Thank you. This was intentional. Those two snippets explain a difference between == and IN. Here's the context from the docs:

Queries that include an equality filter for a given property ignore any sort order specified for that property. For example, the sort order is ignored in the following query:

datastore_eq_query_sorted snippet

This does not apply to queries that include an IN filter. Use the IN operator to combine up to 10 equality (==) clauses on the same property with a logical OR. If you add a sort order for that property, it's applied to the result set.

 datastore_in_query_sorted snippet

**Review by kolea2 — APPROVED** [review]

**Comment by jlara310** [comment]
Looks like ConceptsTest.java runs against the Datastore emulator. The new query operators will not be supported by the emulator any time soon. Not sure how to proceed. Is there a way to run the new tests against a real backend?

**Comment by jlara310** [comment]
Can someone re-trigger the workflows for me again?

**Pull request merged**